sycl/test/basic_tests/accessor/accessor_assignment_operator.cpp

```diff
@@ -0,0 +1,62 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+
+// Testing each of the specialised assignment operators of 0 dimensional
+// buffer accessors, local accessors, and host accessors.
+
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+
+  {
+    int Data = 0;
+
+    using AccT = sycl::accessor<int, 0, sycl::access::mode::read_write>;
+    sycl::buffer<int, 1> Buf(&Data, sycl::range<1>(1));
+
+    Q.submit([&](sycl::handler &CGH) {
+      sycl::accessor<int, 0> A(Buf, CGH);
+      CGH.single_task<class KernelAccessor>([A] {
+        typename AccT::value_type DataToWrite = 1;
+        A = DataToWrite;
+      });
+
+      assert(A == 1);
```

Comment:
Since we do not launch this test and only compile it, I think we can greatly reduce it: it should be enough to only have single `queue::submit` call within which we can test `operator=` for all three types of accessors. There are no need to wait for queue to complete, no need to verify the result, etc.

However, since we do not launch this test, we need to have a E2E test which checks that those new methods actually work. I suggest writing a new test case for intel/llvm-test-suite for regular `accessor` and `local_accessor`. `host_accessor` we can check using unit-tests. As an example of working with host accessor in unit-tests you can take a look at [sycl/unittests/accessor/HostAccessorIterator.cpp](https://github.com/intel/llvm/blob/sycl/sycl/unittests/accessor/HostAccessorIterator.cpp)

Reply:
We could even have a `SYCL_EXTERNAL` function which takes a zero-dimension accessor and makes the assignment.

Reply:
New `host_accessor` unit test [here](https://github.com/intel/llvm/pull/8069/files#diff-d39ed331362bfea08d0e7d92fc0ffa7f155e9ceeb790d85f901ce48e3532da46)

Opened a PR for the E2E tests: https://github.com/intel/llvm-test-suite/pull/1570